index.js

```diff
@@ -155,15 +171,18 @@ class Ky {
 		}, hooks);
 		this._throwHttpErrors = throwHttpErrors;
 
-		const headers = new _globalThis.Headers(this._options.headers || {});
-
 		if (json) {
-			headers.set('content-type', 'application/json');
+			const headers = this._options.headers &&
+				typeof this._options.headers.entries === 'function' ?
+				objectFromEntries(this._options.headers.entries()) :
+				this._options.headers;
+			this._options.headers = {
+				...headers,
+				'content-type': 'application/json'
+			};
```

Comment:
Not a big fan of this. I don't really see the point in adding lots of complications here to save one global. Users needs to polyfill `Response` for Node.js anyway. So adding `Headers` too doesn't change much. The main target is the browser and we should optimize for that.

Reply:
Yeah, that's fair, I will revert this part. My initial intention was to just do this (without the complicated mess above):

```js
this._options.headers = {
    ...this._options.headers,
    'content-type': 'application/json'
};
```
But then I realized that ky supports custom `Headers` instances and those cannot be spread into an object. It would work with object literals, but not `Headers`.

So, fine, I thought, I'll get its entries and convert it. Seems simple enough. Can't use `Object.entries()` on these `Headers` instances, but `headers.entries()` is a thing. Oh, turns out `headers.entries()` doesn't return a normal entries array, but a `Headers Iterator`. Jeez! Well, at least I can spread that into an array and then use `Object.fromEntries()` to make an object out of it. Nope. `Object.fromEntries()` doesn't have good cross-platform support yet.

This shouldn't be so difficult...  😞